AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnScreen.kt

```diff
@@ -0,0 +1,231 @@
+                    AmbientState.Interactive -> R.string.mode_active_label
+                    is AmbientState.Ambient -> R.string.mode_ambient_label
+                }
+            ),
+            style = MaterialTheme.typography.body2,
+            color = stateColor,
+            modifier = Modifier.testTag("mode")
+        )
+        Text(
+            text = stringResource(
+                id = R.string.update_rate_label,
+                when (ambientState) {
+                    AmbientState.Interactive -> ACTIVE_INTERVAL.seconds
+                    is AmbientState.Ambient -> AMBIENT_INTERVAL.seconds
+                }
+            ),
+            style = MaterialTheme.typography.body2,
+            color = stateColor,
+            modifier = Modifier.testTag("rate")
+        )
+        Text(
+            text = stringResource(id = R.string.draw_count_label, drawCount),
+            style = MaterialTheme.typography.body2,
+            color = stateColor,
+            modifier = Modifier.testTag("drawCount")
+        )
+    }
+}
+
+/**
+ * A [Modifier] version of `BoxInsetLayout`. This method assumes that the layout will fill the
+ * entire screen, and that there are no oval devices.
+ */
+private fun Modifier.fillMaxRectangle(): Modifier = composed {
+    val currentWindowMetrics = rememberCurrentWindowMetrics()
+
+    val padding = if (LocalConfiguration.current.isScreenRound) {
+        currentWindowMetrics.width / 2f * (1f - 1f / sqrt(2f))
+    } else {
+        0f
+    }
+
+    this
+        .fillMaxSize()
+        .padding(with(LocalDensity.current) { padding.toDp() })
+}
+
+@Composable
+private fun rememberCurrentWindowMetrics(): Rect {
+    val activity = LocalContext.current.findActivity()
+    val configuration = LocalConfiguration.current
+    return remember(activity, configuration) {
+        WindowMetricsCalculator.getOrCreate().computeCurrentWindowMetrics(activity)
+    }.bounds.toComposeRect()
+}
+
+/**
+ * If the screen requires burn-in protection, items must be shifted around periodically
+ * in ambient mode. To ensure that content isn't shifted off the screen, avoid placing
+ * content within 10 pixels of the edge of the screen.
+ *
+ * Activities should also avoid solid white areas to prevent pixel burn-in. Both of
+ * these requirements only apply in ambient mode, and only when
+ * [AmbientState.Ambient.doBurnInProtection] is set to true.
+ */
+private fun Modifier.burnInTranslation(
+    ambientState: AmbientState,
+    ambientPing: Long
+): Modifier = composed {
+    val translationX = rememberBurnInTranslation(ambientState, ambientPing)
+    val translationY = rememberBurnInTranslation(ambientState, ambientPing)
+
+    this
+        .graphicsLayer {
+            this.translationX = translationX
+            this.translationY = translationY
+        }
+}
+
+@Composable
+private fun rememberBurnInTranslation(ambientState: AmbientState, ambientPing: Long): Float =
+    remember(ambientState, ambientPing) {
```

Comment:
Right, exactly. It's an arbitrary number used to drive the random alignment if needed. We could set the random numbers directly, although that would be a bit awkward from an encapsulation point of view if the ambient handling set values for anything that might need to change when the ambient update happens.

Do you think it would be clearer if we had the `ambientPing` be a timestamp, of the last `ambientUpdate`? We still wouldn't use the value, but that might have more meaning than just an incrementing value.

Reply:
Let me know what you think of the latest approach!

Reply:
LGTM